homeassistant/components/frontier_silicon/__init__.py

```diff
@@ -1 +1,86 @@
-"""The frontier_silicon component."""
+"""The Frontier Silicon integration."""
+from __future__ import annotations
+
+import logging
+
+from afsapi import AFSAPI, ConnectionError as FSConnectionError, InvalidPinException
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PLATFORM,
+    CONF_PORT,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, PlatformNotReady
+from homeassistant.helpers.typing import ConfigType
+
+from .const import CONF_PIN, CONF_WEBFSAPI_URL, DEFAULT_PIN, DEFAULT_PORT, DOMAIN
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Perform migration from YAML config to Config Flow entity."""
+
+    async def _migrate_entry(entry_to_migrate):
+
+        await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_NAME: entry_to_migrate.get(CONF_NAME),
+                CONF_HOST: entry_to_migrate.get(CONF_HOST),
+                CONF_PORT: entry_to_migrate.get(CONF_PORT, DEFAULT_PORT),
+                CONF_PIN: entry_to_migrate.get(CONF_PASSWORD, DEFAULT_PIN),
+            },
+        )
+
+        _LOGGER.warning(
+            "Configuration of the Frontier Silicon integration in YAML is deprecated and "
+            "will be removed in Home Assistant 2022.12; Your existing configuration "
+            "has been imported into the UI automatically and can be safely removed "
+            "from your configuration.yaml file"
+        )
```

Comment:
Best would be to use the new issue / repair function in Home Assistant. See https://github.com/home-assistant/core/pull/80156 as an example.

This has been introduced after you created this PR.